.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,14 @@
+name: pre-commit
+
+on:
+  pull_request:
+  push:
+    branches: [main]
```

Comment:
Is this really pre-commit?

Reply:
Ah, yes! The tool itself is called [pre-commit](https://pre-commit.com/)!

Reply:
Yeah I spotted that, just didn't seem to be an entirely pre-commit use of pre-commit!